dowhy/gcm/falsify.py

```diff
@@ -335,67 +332,80 @@ def validate_causal_minimality(
     )
 
     # Gather results
-    p_values_memory.clear_placeholders()  # Clear placeholders
     for i, (node, p, other_parents) in enumerate(to_test):
         p_values_memory.add_p_value(p_values[i], node, p, other_parents)
 
     # Summarize
     for node, p, other_parents in triples:
-        validation_summary[FALSIFY_N_TESTS] += 1
         p_value = p_values_memory.get_p_value(node, p, other_parents)
-        validation_summary[FALSIFY_P_VALUES][(node, p, tuple(other_parents))] = (p_value, p_value > significance_level)
-        if p_value > significance_level:
-            validation_summary[FALSIFY_N_VIOLATIONS] += 1
+        if p_value != -1:
+            validation_summary[FalsifyConst.N_TESTS] += 1
+            validation_summary[FalsifyConst.P_VALUES][(node, p, tuple(other_parents))] = (
+                p_value,
+                p_value > significance_level,
+            )
+            if p_value > significance_level:
+                validation_summary[FalsifyConst.N_VIOLATIONS] += 1
 
     return validation_summary
 
 
 def validate_graph(
     causal_graph: DirectedGraph,
-    data: pd.DataFrame,
-    methods: Union[Callable, Tuple[Callable, ...], List[Callable]] = (validate_lmc, validate_pd),
-    independence_test: Callable[[np.ndarray, np.ndarray], float] = kernel_based,
-    conditional_independence_test: Callable[[np.ndarray, np.ndarray, np.ndarray], float] = kernel_based,
+    data: Optional[pd.DataFrame] = None,
+    methods: Optional[Union[Callable, Tuple[Callable, ...], List[Callable]]] = None,
```

Comment:
What about the following alternative: Remove all the method specific parameters here and set a default method partial function. This is, something like:

```
def validate_graph(
   causal_graph: DirectedGraph,
   methods=(
      wrap_partial(validate_lmc, data=data, independence_test=kernel_based, conditional_independence_test=kernel_based)
      ),
   significance_level: float = 0.05,
   n_jobs: Optional[int] = None
)
```
Then there is no need to have these mix of parameters for the default behavior and customization, since now the default behavior is also defined as a partial.